File: Striver_SDE_SHEET/646-maximum-length-of-pair-chain/maximum-length-of-pair-chain.cpp
class Solution {
public:
    int findLongestChain(vector<vector<int>>& pairs) {
        int n = pairs.size();
        sort(pairs.begin(), pairs.end(), [](const std::vector<int>& a, const std::vector<int>& b) {
            return a[1] < b[1]; 
        });
        int prev = pairs[0][1], chains=1;
        for(int i=1;i<n;i++){
            if(pairs[i][0]>prev){
                chains+=1;
                prev = pairs[i][1];
            }
        }
        return chains;
    }
};

File: Striver_SDE_SHEET/646-maximum-length-of-pair-chain/maximum-length-of-pair-chain.py
class Solution:
    def findLongestChain(self, pairs: List[List[int]]) -> int:
        pairs.sort(key = lambda x: x[-1])
        count = 0
        prev = pairs[0][1]
        for s,e in pairs[1:]:
            if s>prev:
                count+=1
                prev = e
        return count+1
